### PR TITLE
WIP: Add -m 11500 module & unit test

### PR DESCRIPTION
**Pull request by mhasbini**
Still investigating test failures.

### CODE REVIEW

File: src/modules/module_11500.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+  if (parse_rc == false) return (PARSER_SALT_LENGTH);
+
+  return (PARSER_OK);
+}
+
+int module_hash_encode (MAYBE_UNUSED const hashconfig_t *hashconfig, MAYBE_UNUSED const void *digest_buf, MAYBE_UNUSED const salt_t *salt, MAYBE_UNUSED const void *esalt_buf, MAYBE_UNUSED const void *hook_salt_buf, MAYBE_UNUSED const hashinfo_t *hash_info, char *line_buf, MAYBE_UNUSED const int line_size)
+{
+  const u32 *digest = (const u32 *) digest_buf;
+
+  const int line_len = snprintf (line_buf, line_size, "%08x:%s", digest[0], (char *) salt->salt_buf);
```
Comment (mhasbini) [inline src/modules/module_11500.c]: @jsteube `digest[0] ` was originally `byte_swap_32 (digest_buf[0])`, but this produced erroneous hashes. e.g. for input `309273c2:00000000` it gave: `c2739230:`. Thus I removed the swap, although not sure if I'm missing something here.

File: tools/test_modules/m11500.pm [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+use Digest::CRC qw (crc32);
+
+sub module_constraints { [[0, 255], [8, 8], [0, 31], [8, 8], [-1, -1]] }
+
+sub module_generate_hash
+{
+  my $word = shift;
+  my $salt = shift;
+
+  my $digest = crc32 ($word);
```
Comment (mhasbini) [inline tools/test_modules/m11500.pm:19]: @jsteube I'm not sure if I should concat `$salt` here or not.

Comment (jsteube) [reply tools/test_modules/m11500.pm:19]: The perl module does not work as expected, therefore we just set it to fixed zero. You can remove it and use sprintf ("%08x:00000000", $digest);

File: src/modules/module_11500.c (hunk 1 above)
Comment (jsteube) [reply src/modules/module_11500.c]: This should do it:

const int line_len = snprintf (line_buf, line_size, "%08x:%08x", digest[0], salt->salt_buf[0]);

Comment (mhasbini) [reply src/modules/module_11500.c]: Thanks :+1:

**Pull request merged**